From 77d8e9fa07caf7d0930521a07dc5565f0d56e1a4 Mon Sep 17 00:00:00 2001
From: "guoyin.chen" <guoyin.chen@freescale.com>
Date: Mon, 7 May 2012 15:42:32 +0800
Subject: [PATCH 25/38] ENGR00182086 Remove the delay init for reference-ril
 library

RIL framework will handle the error if 3g modem is not ready to be used

Signed-off-by: guoyin.chen <guoyin.chen@freescale.com>
---
 reference-ril/reference-ril.c | 19 +++----------------
 1 file changed, 3 insertions(+), 16 deletions(-)

diff --git a/reference-ril/reference-ril.c b/reference-ril/reference-ril.c
index 016ebd4..6a6adaa 100755
--- a/reference-ril/reference-ril.c
+++ b/reference-ril/reference-ril.c
@@ -3712,22 +3712,9 @@ mainLoop(void *param)
             return 0;
         }
 
-        /* Some HUAWEI USB datacards (e.g. E180) bootup and attach to network a little bit slowly(e.g. 20s). 
-           Delay RIL init so that we can always get correct network status (i.e. +COPS?).
-           FIX ME: this should be removed in future. Service layer should be responsible to handle "late" attach.
-        */
-        property_get(RIL_DELAY_TIME, delay_init, "0");
-        delay = strtol(delay_init, NULL, 10);
-        if(delay > 120) {
-            LOGW ("Should not delay RIL init too long (%d seconds). Please set %s properly", delay, RIL_DELAY_TIME);
-            delay = 0;
-        }
-        if (current_modem_type == HUAWEI_MODEM) {
-            TIMEVAL_DELAYINIT.tv_sec = delay;
-            LOGD ("Will delay RIL initialization for %d seconds", TIMEVAL_DELAYINIT.tv_sec); 
-        } else			/* Found other modem don't need delay */
-            TIMEVAL_DELAYINIT.tv_sec = 0;
-
+        //RIL framework can handle the modem not response on AT+COPS in time
+        //So remove the delay here
+        TIMEVAL_DELAYINIT.tv_sec = 0;
 
         RIL_requestTimedCallback(initializeCallback, NULL, &TIMEVAL_DELAYINIT);
 
-- 
1.8.0

